cmd/topicmappr/commands/scale.go

```diff
@@ -92,184 +52,13 @@ func scale(cmd *cobra.Command, _ []string) {
 
 	defer zk.Close()
 
-	// Get broker and partition metadata.
-	if err := checkMetaAge(zk, params.maxMetadataAge); err != nil {
-		fmt.Println(err)
-		os.Exit(1)
-	}
-	brokerMeta, errs := getBrokerMeta(zk, true)
-	if errs != nil && brokerMeta == nil {
-		for _, e := range errs {
-			fmt.Println(e)
-		}
-		os.Exit(1)
-	}
-	partitionMeta, err := getPartitionMeta(zk)
-	if err != nil {
-		fmt.Println(err)
-		os.Exit(1)
-	}
-
-	// Get the current partition map.
-	partitionMapIn, err := kafkazk.PartitionMapFromZK(params.topics, zk)
-	if err != nil {
-		fmt.Println(err)
-		os.Exit(1)
-	}
-
-	// Exclude any topics that are pending deletion.
-	pending, err := stripPendingDeletes(partitionMapIn, zk)
-	if err != nil {
-		fmt.Println("Error fetching topics pending deletion")
-	}
-
-	// Exclude any explicit exclusions.
-	excluded := removeTopics(partitionMapIn, params.topicsExclude)
-
-	// Print topics matched to input params.
-	printTopics(partitionMapIn)
-
-	// Print if any topics were excluded due to pending deletion.
-	printExcludedTopics(pending, excluded)
-
-	// Get a broker map.
-	brokersIn := kafkazk.BrokerMapFromPartitionMap(partitionMapIn, brokerMeta, false)
-
-	// Validate all broker params, get a copy of the
-	// broker IDs targeted for partition offloading.
-	offloadTargets := validateBrokersForScale(params, brokersIn, brokerMeta)
-
-	// Sort offloadTargets by storage free ascending.
-	sort.Sort(offloadTargetsBySize{t: offloadTargets, bm: brokersIn})
-
-	// Generate reassignmentBundles for a scale up.
-	reassignmentBundlesParams := computeReassignmentBundlesParams{
-		offloadTargets:         offloadTargets,
-		tolerance:              params.tolerance,
-		partitionMap:           partitionMapIn,
-		partitionMeta:          partitionMeta,
-		brokerMap:              brokersIn,
-		partitionLimit:         params.partitionLimit,
-		partitionSizeThreshold: params.partitionSizeThreshold,
-		localityScoped:         params.localityScoped,
-		verbose:                params.verbose,
-	}
-
-	results := computeReassignmentBundles(reassignmentBundlesParams)
-
-	// Merge all results into a slice.
-	resultsByRange := []reassignmentBundle{}
-	for r := range results {
-		resultsByRange = append(resultsByRange, r)
-	}
+	partitionMap, _ := reassign(params, zk)
 
-	// Sort the scale results by range ascending.
-	sort.Slice(resultsByRange, func(i, j int) bool {
-		switch {
-		case resultsByRange[i].storageRange < resultsByRange[j].storageRange:
-			return true
-		case resultsByRange[i].storageRange > resultsByRange[j].storageRange:
-			return false
-		}
+	// TODO intentionally not handling the one error that can be returned here
```

Comment:
I saw that the range increase error is always returned from `printBrokerAssignmentStats`; is the goal here to conditionally handle it here rather than in that function I assume?